### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by Khaled-Alaa-1**
### In this project we :

- Use React Router.
- Add root pages
   -Use styles in a React app.

### CODE REVIEW

File: src/logic/operate.test.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import operate from './operate';
+
+describe('operate', () => {
+  it('should return the sum of two numbers when passed "+" as the operation', () => {
+    const result = operate('5', '3', '+');
+    expect(result).toEqual('8');
+  });
+});
```
Comment (DuaneDave) [inline src/logic/operate.test.js]: According to the requirements, tests that cover all scenarios of the operate js should be written and I recommend writing more tests to see if other functions are working as expected.

File: src/logic/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import calculate from './calculate';
+
+describe('calculate', () => {
+  it('should return an object with total, next, and operation properties when passed an AC buttonName', () => {
+    const result = calculate({ total: '5', next: '3', operation: '+' }, 'AC');
+    expect(result).toEqual({ total: null, next: null, operation: null });
+  });
+});
```
Comment (DuaneDave) [inline src/logic/calculate.test.js]: According to the requirements, tests that cover all scenarios of the calculate js should be written and I recommend writing more tests to see if other functions are working as expected.

**Review by DuaneDave — CHANGES_REQUESTED** [review]
Hi @Khaled-Alaa-1 and @PedroH-Neves 👋 , 

Good job so far on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Correct usage of git-flow
✔️ Good usage of react testing library

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @Khaled-Alaa-1 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented requested changes to ensure all tests scenario are covered and all tests are passing.

## Optional suggestions

- [x] Great work implementing all changes requested I would recommend you add tests for all components.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**